src/main/java/io/quarkus/search/app/hibernate/AnalysisConfigurer.java

```diff
@@ -18,6 +26,17 @@ public void configure(ElasticsearchAnalysisConfigurationContext context) {
                 .tokenizer("standard")
                 .tokenFilters("lowercase", "asciifolding", "stemmer")
                 .charFilters("html_strip");
+
+        context.analyzer(DEFAULT_SEARCH).custom()
+                .tokenizer("standard")
+                .tokenFilters("lowercase", "asciifolding", "stemmer", "synonyms_graph_filter")
```

Comment:
Makes sense to apply the synonyms after the stemmer, but... did you check that `service` isn't "stemmed down" to just `servic`? In which case the synonyms might not be working as expected.

Reply:
As far as I understand, the filters before the synonym on should get applied to the list of synonyms themselves, so that was why I've added it at the end. From https://www.elastic.co/blog/boosting-the-power-of-elasticsearch-with-synonyms see `Being a part of the (analysis) chain gang`:

> In general, synonym filters rewrite their inputs to the tokenizer and filters used in the preceding analysis chain

I'll add a comment about this to the config too.